docs/help/contributor/style-guide/documenting-code.md

```diff
@@ -0,0 +1,212 @@
+Use **kubectl edit** for files which are shipped as part of the Knative software, like the serving/eventing ConfigMaps.
+
+=== ":white_check_mark: Correct"
+
+    Creating a new file:
+    ```
+    kubectl apply -f - <<EOF
+    # code
+    EOF
+    ```
+
+    Editing a file:
+    ```
+    kubectl -n <namespace> edit cm <filename>
+    ```
+
+=== ":no_entry: Incorrect"
+    ```
+    cat <<EOF | kubectl create -f -
+    # code
+    EOF
+    ```
+
+
+## Words requiring code formatting
+Use code formatting to indicate special purpose text. Apply code formatting to the following content:
+
+* Text that the user enters into a field
+* Filenames and path names
+* Configuration file content
+* Sections of code
+* Programming examples
+* Command line text
+* Environment variable text
+
+
+## Referencing Variables in Code Blocks
+
+>Format variables in code blocks like so: <service-name>
+
+> - All lower case
+- Hyphens between words
+- Explanation for each variable below code block
+- Explanation format is “Where... `<service-name>` is…"
+- If there are multiple variables, see below.
+
+### Single Variable
+=== ":white_check_mark: Correct"
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+=== ":no_entry: Incorrect"
+    ```bash
+    kn create service {SERVICE_NAME}
+    ```
+    {SERVICE_NAME} = The name of your service
+
+
+### Multiple Variables
+
+=== ":white_check_mark: Correct"
+    ```bash
+    kn create service <service-name> --revision-name <revision-name>
+    ```
+    Where:
+
+    * `<service-name>` is the name of your Knative Service.
+    * `<revision-name>` is the desired name of your revision
+
+=== ":no_entry: Incorrect"
+    ```bash
+    kn create service <service-name> --revision-name <revision-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.<br>
+    Where `<revision-name>` is the desired name of your revision
+
+
+## CLI Input and Output
+### CLI Input
+> Be sure to specify the language your code is in, bash for any CLI commands.
+
+=== ":white_check_mark: Correct"
+    ```go
+    package main
+
+    import "fmt"
+
+    func main() {
+        fmt.Println("hello world")
+    }
+    ```
+
+=== ":no_entry: Incorrect"
+    ```bash
+    package main
+
+    import "fmt"
+
+    func main() {
+        fmt.Println("hello world")
+    }
+    ```
+
+=== ":white_check_mark: Correct Formatting"
+    ````
+    ```go
+    package main
+
+    import "fmt"
+
+    func main() {
+        fmt.Println("hello world")
+    }
+    ```
+    ````
+=== ":no_entry: Incorrect Formatting"
+    ````
+    ```bash
+    package main
+
+    import "fmt"
+
+    func main() {
+        fmt.Println("hello world")
+    }
+    ```
+    ````
+
+### CLI Output
+> CLI Output should include the custom css "{ .bash .no-copy }" in place of "bash" which removes the "Copy to clipboard button" on the right side of the code block
```

Comment:
Can you call out somewhere that for terminal stuff that isn't a specific language we should use `bash`? Also, for the `kubectl apply` type commands I think we should use `yaml` as the language since we would want the bit that's being applied in YAML syntax highlighting rather than `bash`?

Reply:
yup, on it